app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewsController < ApplicationController
+    before_action :authorize_user!, only: [:destroy]
+    
+    def new
+        @review = Review.new
+    end
+
+    def create
+        @review = Review.new review_params
+        @review.user = current_user
+        @idea = Idea.find params[:idea_id]
+        @review.idea = @idea
+        if @review.save review_params
+            redirect_to idea_path(@idea.id)
+        else
+            @reviews = @idea.reviews
+            render "ideas/show"
+        end
+    end
+
+    def destroy
+        @review = Review.find params[:id]
+        @review.destroy
+        redirect_to idea_path(@review.idea)
+    end
+
+    private
+
+    def review_params
+        params.require(:review).permit(:title)
+    end
+
+    def authorize_user!
+        unless can?(:crud, @review)
```

Comment:
The problem with authorize_user! method here is that @review is nil. You forgot to query it `@review = Review.find(params.require(:id).permit(:id))` within the method OR better practice define a method in app/models/review which can then be evoked on the before_action filter

> app/controllers/reviews_controller.rb
```ruby
before_action Review.get_review_from_params
```

>app/models/review
```ruby
def get_review_from_params
@review = Review.find(params.require(:id).permit(:id))
end
```